server/src/main/java/com/battleships/PlayerCommand.java

```diff
@@ -1,21 +1,32 @@
 package com.battleships;
 
+import java.util.Optional;
+
 class PlayerCommand {
     private final Command type;
-    private final String value;
+    private Optional<String> valueOptional;
 
     PlayerCommand(String playerInput) {
         // TODO validate it !
         String[] inputCommand = playerInput.split(CommunicatingProtocol.getSeparator());
```

Comment:
dot train. Extract separator before using it as split method parameter.

Reply:
extracted